src/StoreApi/Schemas/CartSchema.php

```diff
@@ -356,6 +362,7 @@ public function get_item_response( $cart ) {
 				]
 			),
 			'errors'                  => $cart_errors,
+			'payment_requirements'    => apply_filters( 'woocommerce_blocks_store_api_cart_payment_requirements', array( 'products' ) ),
```

Comment:
I don't think we should be using filters here, we should use ExtendRestAPI, here's why:
**- Filters code can break**: any exception would bubble up to our cart response and return a broken response (breaking the cart).
**- Filters can break other plugins**: developers might forget to append to the array passed, might hardcode an array, causing the code applied by other plugins to be superseded.
**- Filters are not type-safe**: developers can return a string (instead of an array of string), return an object, or any other shape that might break our frontend integration of it, causing all payment gateways to not works.

Generally, when developing new integration points, we go with the assumption that if a developer can break code via out API, then eventually he will.

ExtendRestAPI is a glorified filter, that simply tries to fix the above problems, without using filters, because if you use filters, people would bypass your code and directly inject into the filter.

Any integration point we introduce should be:
- Lazy loaded, only executing the developer code once we need it.
- Wrapped in a try-catch, to ignore and remove the code if it has errors and only surface the error to admins.
- Type checked, so that simple issues are caught early, type errors should trigger an exception that is handled like the above.

Reply:
I understand the concerns and I know what issues are associated with using filters. I do honestly believe that we should not use ExtendRestAPI in such a simple scenario. This is overkill. That being said I rely on your judgment and will change the implementation.

So your proposal means that that the per extension requirement will be part of the checkout endpoint response. In the case of subscriptions it will be:
`extensions->subscriptions->payment_requirements: []` Do you think that we should indicate that a particular extension has requirements or should we parse all entries in extensions and look for a particular `key`? All the items from different plugins should be combined into one value. All of this should happen inside `useStoreCart`. Is that correct?

Reply:
> checkout endpoint response. In the case of subscriptions it will be: `extensions->subscriptions->payment_requirements: []`

No, the data will always stay here as is, the API could be shaped like this

```
$extend->register_payment_requirements( [ 'subscriptions', 'multiple_ subscriptions' ] );
```

Reply:
so instead of people calling the filter, they call the above code, have in run inside `woocommerce_blocks_loaded` action or something, just like the rest of API.

Reply:
Done. I used your suggestion from Slack about how to shape this. It does not care about the namespace or the endpoint. I don't think that those parameters matter in this scenario. Should they be added just in case?